### PR TITLE
refactor: cron job gd exports error after update

### PR DESCRIPTION
**Pull request by JBR90**
## Description

This is for the cron job that runs overnight and deletes google drive files that have expired (older than 2 weeks)

- Moves error outside of loop, reports error with all file ids that have failed in one error.

## Checklist

- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts [hunk 1]
```diff
@@ -54,14 +54,13 @@ async function updateExpiredAtAndDelete(fileIds: string[]) {
...
-      log.error(errorMessage);
-      throw new Error(errorMessage);
-    }
+  }
+  if (failedIds.length > 0) {
+    const errorMessage = `Failed to process the following file IDs: ${failedIds.join(
+      ", ",
+    )}`;
+    log.error(errorMessage);
+    throw new Error(errorMessage);
```
Comment (codeincontext) [inline apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:63]: Could this potentially be 1000 IDs? I don't know if there's a limit on an error message length, but I'm happy to go with it and we'll see if it needs another tweak or not

**Review by codeincontext — APPROVED** [review]

File: packages/api/src/router/testSupport/prepareUser.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ const generateEmailAddress = (personaName: keyof typeof personas) => {
     // All users use the "test@thenational.academy" mailbox with a `+` alias
     "test",
     // Replace non-alphanumeric characters with -
-    branch.replace(/\W+/g, "-"),
+    branch.replace(/\W+/g, "-").slice(-15),
```
Comment (JBR90) [inline packages/api/src/router/testSupport/prepareUser.ts:98]: @codeincontext - bit of a crude solution here. Will this cause uniqueness issues?

Comment (codeincontext) [reply packages/api/src/router/testSupport/prepareUser.ts:98]: Let's go for it and see what happens. I think it'll be fine

**Review by codeincontext — APPROVED** [review]

**Pull request merged**